lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -120,21 +138,30 @@ static LogicalResult resolveClassStructBody(ClassDeclOp op,
 
   if (auto baseClass = op.getBaseAttr()) {
 
-    // Process base class' struct layout first
-    auto baseClassStruct = cache.getStructInfo(baseClass);
-    if (!baseClassStruct)
-      return op.emitOpError() << "Base class " << baseClass << " of "
-                              << classSym << " has not been converted.";
-
-    structBodyMembers.push_back(baseClassStruct->classBody);
-    derivedStartIdx = 1;
-
-    // Inherit base field paths with a leading 0.
-    for (auto &kv : baseClassStruct->propertyPath) {
-      SmallVector<unsigned, 2> path;
-      path.push_back(0); // into base subobject
-      path.append(kv.second.begin(), kv.second.end());
-      structBody.setFieldPath(kv.first, path);
+    ModuleOp mod = op->getParentOfType<ModuleOp>();
+    // It's possible the base class does not resolve since it materialzes
+    // no methods or properties. In that case we can simply ignore inheritance.
+    if (auto *opSym = mod.lookupSymbol(baseClass)) {
```

Comment:
Shouldn't this be an error? If the class mentions a base class symbol, we should enforce that symbol to exist. The symbol user verifier should already do this. That means you can just `assert(opSym)` here :smiley: